src/content/learn/web-app-architectures.md

```diff
@@ -0,0 +1,290 @@
+
+
+## Why We Recommend Frameworks {/*why-we-recommend-frameworks*/}
+
+Although you could solve all these pieces yourself in a build tool like Create React App, Vite, or Parcel, it is hard to do well. Just like when Create React App itself integrated several build tools together, you need a tool to integrate all of these features together to provide the best experience to users.
+
+This category of tools that integrates build tools, rendering, routing, data fetching, and code splitting are known as "frameworks" -- or if you prefer to call React itself a framework, you might call them "metaframeworks".
+
+Frameworks impose some opinions about structuring your app in order to provide a much better user experience, in the same way build tools impose some opinions to make tooling easier. This is why we started recommending frameworks like [Next.js](https://nextjs.org/), [React Router](https://reactrouter.com/), and [Expo](https://expo.dev/) for new projects.
+
+Frameworks provide the same getting started experience as Create React App, but also provide solutions to problems users need to solve anyway in real production apps.
+
+## Architecture Considerations {/*architecture-considerations*/}
+
+There are many factors that go into deciding the right architecture(s) for each application.  Here are some important points to take into consideration.
+
+### Application Types {/*application-types*/}
+
+One way to approach deciding on the right architectures is to look at the kind of application you are building.
+
+The post [Application Holotypes: A Guide to Architecture Decisions](https://jasonformat.com/application-holotypes/) describes a spectrum of applications, each with their own variation of architectural needs. For example, a social media app might work well as a Single Page App, whereas an ecommerce site might benefit from a Server-Side Rendering architecture with some client interactivity.
+
+The [Patterns.dev guide on rendering and architectural patterns](https://www.patterns.dev/vanilla/rendering-patterns) also gives excellent insight into the different architectural techniques available and when to apply them.
+
+### Server rendering is optional {/*server-rendering-is-optional*/}
+
+The frameworks we recommend all provide the option to create a [client-side rendered (CSR)](https://developer.mozilla.org/en-US/docs/Glossary/CSR) app.
+
+In some cases, CSR is the right choice for a page, but many times it's not. Even if most of your app is client-side, there are often individual pages that could benefit from server rendering features like [static-site generation (SSG)](https://developer.mozilla.org/en-US/docs/Glossary/SSG) or [server-side rendering (SSR)](https://developer.mozilla.org/en-US/docs/Glossary/SSR), for example a Terms of Service page, or documentation.
```

Comment:
I'm not sure if I understand this correctly, but imho there is a difference between SSG and SSR in the way that for SSR you need a server at runtime and SSG only at build time. While this is clarified above in `Static Site Generation (SSG)´ section, it's somehow mixed here.

Also the "Terms of Service page" example is a little weak imho. Would you really add server infrastructure only for a ToS page (assuming you don't have JS on the server already)? :-)